packages/graphql/src/services/graphql.service.ts

```diff
@@ -195,7 +195,7 @@ export class GraphqlService implements BackendService {
 
     const set = (o: any = {}, a: any) => {
       const k = a.shift();
-      o[k] = a.length ? set(o[k], a) : null;
+      o[k] = a.length ? set(o[k] ?? undefined, a) : null;
```

Comment:
from your description, you say doing this would force the use of the default `{}` but in that case why not provide it directly here instead? Wouldn't it be better to simply use `set(o[k] ?? {}, a)`

Reply:
Sure, I can do that if you prefer.

However, the benefit of passing `undefined`, is that if you decide to change the default value to something else in the future, you won't have to update the places it's passed in.
eg:
Changing the signature to
`const set = (o: any = { a: true }, a: any)`
Means updating
`set(o[k] ?? { a: true }, a)` // to match the new default value
While, doesn't need to change
`set(o[k] ?? undefined, a)` // o will end up having the update default value as specified

Reply:
I don't see why it would ever change and I also think passing `{}` is much more obvious because I would not have imagined that it would use the default until you said it would, it's less clear with `undefined` compare to `{}` which is clear what it does